python/setup.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, Blue Brain Project
+#                     Daniel Nachbaur <daniel.nachbaur@epfl.ch>
+#
+# This file is part of Rockets <https://github.com/BlueBrain/Rockets>
+#
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+# All rights reserved. Do not distribute without further notice.
+
+"""setup.py"""
+import os
+from setuptools import setup
+
+try: # for pip >= 10
+    from pip._internal.req import parse_requirements
+    from pip._internal.download import PipSession
+except ImportError: # for pip <= 9.0.3
+    from pip.req import parse_requirements
+    from pip.download import PipSession
+from optparse import Option
+
+BASEDIR = os.path.dirname(os.path.abspath(__file__))
+
+
+def parse_reqs(reqs_file):
+    ''' parse the requirements '''
+    options = Option("--workaround")
+    options.skip_requirements_regex = None
+    options.isolated_mode = True
+    install_reqs = parse_requirements(reqs_file, options=options, session=PipSession())
+    return [str(ir.req) for ir in install_reqs]
```

Comment:
How difficult would it be to replace all the pip stuff to parse requirements with your own future proof code? This looks already like a lot of code, and parsing the requirements file 
I remember reading a comment from a pip developer saying this was a bad idea and parsing the requirements file seems simpler that all this code here.

Reply:
fair enough